pkg/services/user/userimpl/user.go

```diff
@@ -467,89 +468,42 @@ func (s *Service) getOrgIDForNewUser(ctx context.Context, cmd *user.CreateUserCo
 	return orgID, err
 }
 
-// CreateServiceAccount is a copy of Create with a single difference; it will create the OrgUser service account.
+// CreateServiceAccount creates a service account in the user table and adds service account to an organisation in the org_user table
 func (s *Service) CreateServiceAccount(ctx context.Context, cmd *user.CreateUserCommand) (*user.User, error) {
-	cmdOrg := org.GetOrgIDForNewUserCommand{
-		Email:        cmd.Email,
-		Login:        cmd.Login,
-		OrgID:        cmd.OrgID,
-		OrgName:      cmd.OrgName,
-		SkipOrgSetup: cmd.SkipOrgSetup,
-	}
-	orgID, err := s.orgService.GetIDForNewUser(ctx, cmdOrg)
+	cmd.Email = cmd.Login
+	err := s.store.LoginConflict(ctx, cmd.Login, cmd.Email, s.cfg.CaseInsensitiveLogin)
 	if err != nil {
-		return nil, err
-	}
-	if cmd.Email == "" {
-		cmd.Email = cmd.Login
-	}
-
-	err = s.store.LoginConflict(ctx, cmd.Login, cmd.Email, s.cfg.CaseInsensitiveLogin)
-	if err != nil {
-		return nil, err
+		return nil, serviceaccounts.ErrServiceAccountAlreadyExists
 	}
 
 	// create user
 	usr := &user.User{
 		Email:            cmd.Email,
 		Name:             cmd.Name,
 		Login:            cmd.Login,
-		Company:          cmd.Company,
-		IsAdmin:          cmd.IsAdmin,
 		IsDisabled:       cmd.IsDisabled,
 		OrgID:            cmd.OrgID,
-		EmailVerified:    cmd.EmailVerified,
 		Created:          time.Now(),
 		Updated:          time.Now(),
 		LastSeenAt:       time.Now().AddDate(-10, 0, 0),
-		IsServiceAccount: cmd.IsServiceAccount,
+		IsServiceAccount: true,
 	}
-
-	salt, err := util.GetRandomString(10)
-	if err != nil {
-		return nil, err
-	}
-	usr.Salt = salt
-	rands, err := util.GetRandomString(10)
+	_, err = s.store.Insert(ctx, usr)
 	if err != nil {
 		return nil, err
 	}
-	usr.Rands = rands
```

Comment:
I think it's safe to remove these, as service account should never need them. But please correct me if I'm wrong.

Reply:
(I'm lacking context around all this, so _grain of salt_ 😅...)

don't we need `Salt` for auth? it's referenced [here](https://github.com/grafana/grafana/blob/b44b6fc5c6ca032988f4a17b2e2932f3b89f32b7/pkg/services/authn/clients/grafana.go#L100) - or is that auth method not used for SA users?

also `Rands` is used for password resetting (and nothing else from what I can tell) - maybe that should be explicitly disallowed for SA users?

Reply:
We should never use a password login for service accounts, they should only authenticate using a service account token. However, there might still be some logic that relies on user object always having salt and rands set.

@Jguer and @kalleep do you know if it would be safe to remove these? Or should I leave them in just to be on the safe side.

Reply:
As long as service accounts are safely disabled from basic auth, it should be safe to remove these fields. 

(I went and checked around the code and could not find these being used for anything else that's not password auth)

Reply:
(I'm just afraid some service in the future tries to use salted hashing for something in the future and does not take into account SAs)

Reply:
Fair point. I think I'll put them back in just to be on the safe side.